Screenmarkers plugin redesign

psikoi opened the pull request:
I'm sorry, it's 8h30m in the morning and I have been coding this for
7 straight hours, I underestimated the size of this commit.
I can't think clearly enough to write a proper commit or PR message.

I would ask of the other developers that were involved in this project already to please finish this PR.
@Kamielvf  @deathbeam  @Adam- 

TO-DO list:
- (Maybe) add a PluginErrorPanel explaining what screen markers are if the list is empty (see implementations on ExpTrackers or Ge)
- Hover effects, different shade versions of icons as an hover effect
- Only allow creation of markers when logged in, and if player logs out in the process of creating one, dismiss the creation mode

I'm sorry if the code isn't up to my latest standarts, just trying to get this out before thursday, would love some patches from other devs.

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/screenmarkers/ScreenMarkerOverlay.java:
```diff
@@ -47,6 +47,11 @@
+
+	public void update()
+	{
 		screenMarkerRenderable = new ScreenMarkerRenderable();
```
Instead of this I would just set all the properties in the render method, and leave the renderable final and initialize when declaring

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/screenmarkers/ui/ScreenMarkerPanel.java:
```diff
@@ -24,55 +25,379 @@
+	private boolean visible;
+	private boolean editingThickness;
+
+	private final JLabel borderColorIndicator = new JLabel();
```
Move final declarations before non-final declarations

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/screenmarkers/ui/ScreenMarkerPanel.java:
```diff
@@ -24,55 +25,379 @@
+				{
+					marker.getMarker().setFill(new Color(fill.getRed(), fill.getGreen(), fill.getBlue(), 0));
+				}
+				updateFill();
```
Add empty line above

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/screenmarkers/ui/ScreenMarkerPanel.java:
```diff
@@ -24,55 +25,379 @@
+			public void mousePressed(MouseEvent mouseEvent)
+			{
+				Color fill = marker.getMarker().getFill();
+				if (marker.getMarker().isFullyTransparent())
```
Add empty line above

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/screenmarkers/ui/ScreenMarkerPanel.java:
```diff
@@ -24,55 +25,379 @@
+				int confirm = JOptionPane.showConfirmDialog(null,
+					"Are you sure you want to permanenely delete this screen marker?",
+					"Warning", JOptionPane.OK_CANCEL_OPTION);
+				if (confirm == 0)
```
Add empty line above

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/screenmarkers/ui/ScreenMarkerPanel.java:
```diff
@@ -24,55 +25,379 @@
+			Color fullColor = new Color(color.getRed(), color.getGreen(), color.getBlue());
+			fillColorIndicator.setBorder(new MatteBorder(0, 0, 3, 0, fullColor));
+		}
+		updateOpacity();
```
Add empty line above

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/screenmarkers/ui/ScreenMarkerPanel.java:
```diff
@@ -24,55 +25,379 @@
+			Color color = marker.getMarker().getColor();
+			borderColorIndicator.setBorder(new MatteBorder(0, 0, 3, 0, color));
+		}
+		borderColorIndicator.setIcon(marker.getMarker().getBorderThickness() == 0 ? NO_BORDER_COLOR_ICON : BORDER_COLOR_ICON);
```
Add empty line above

j03l commented:
If I could add my two cents, the first character of the marker name is chopped off no matter how much you resize. 
![image](https://user-images.githubusercontent.com/4055766/40424577-524dd176-5ed9-11e8-941d-46a9250f68b6.png)

Kamielvf commented:
Been testing your changes, if you start creating one and press right mouse button, it removes the creation panel on the side but still allows you to draw screen markers.

psikoi commented:
@j03l that is not in this PR, that's the outdated version I'm trying to update

psikoi commented:
- Fixed @deathbeam's review requests
- Added Hover effects
- Added tooltips
- Added introduction text panel
- Fixed right click to cancel

deathbeam commented on runelite-client/src/main/java/net/runelite/client/plugins/screenmarkers/ui/ScreenMarkerPluginPanel.java:
```diff
@@ -28,184 +29,207 @@
+	private static final Color DEFAULT_FILL_COLOR = new Color(0, 255, 0, 0);
+
+	private static final int DEFAULT_BORDER_THICKNESS = 3;
+	public static final int DEFAULT_FILL_OPACITY = 75;
```
This is i think the last thing missing here. We should get this into this release imo

psikoi commented:
- Moved a constant to a more appropriate class.

@deathbeam we gud

psikoi commented:
- Fixed bug with introduction panel appearing when it shouldn't
- Made the thickness editor only save to config when exiting to display mode
- Changed click listener on panels to focus lost listener on the spinner to save

psikoi commented:
- Removed click-outside to save
- Fixed enter keypress to save
